src/lib.rs

```diff
@@ -1,172 +1,172 @@
-mod abi;
-mod pb;
 use hex_literal::hex;
-use pb::contract::v1 as contract;
+use substreams::errors::Error;
 use substreams::Hex;
-use substreams_database_change::pb::database::DatabaseChanges;
-use substreams_database_change::tables::Tables as DatabaseChangeTables;
-use substreams_entity_change::pb::entity::EntityChanges;
-use substreams_entity_change::tables::Tables as EntityChangesTables;
-use substreams_ethereum::pb::eth::v2 as eth;
-use substreams_ethereum::Event;
-
-#[allow(unused_imports)]
-use num_traits::cast::ToPrimitive;
-use std::str::FromStr;
-use substreams::scalar::BigDecimal;
+use substreams_ethereum::pb::eth::v2::{self as eth};
+
+mod abi;
+mod pb;
+
+use abi::registry::events::{
+    BasePoolAdded, CryptoPoolDeployed, MetaPoolDeployed, PlainPoolDeployed,
+    PoolAdded1 as RegistryPoolAdded, PoolAdded2 as CryptoSwapPoolAdded,
+};
+use pb::curve::types::v1::{Pool, Pools};
 
 substreams_ethereum::init!();
 
-const POOLREGISTRY_TRACKED_CONTRACT: [u8; 20] = hex!("90e00ace148ca3b23ac1bc8c240c2a7dd9c2d7f5");
-const POOLREGISTRY2_TRACKED_CONTRACT: [u8; 20] = hex!("7d86446ddb609ed0f5f8684acf30380a356b2b4c");
-
-fn map_poolregistry_events(blk: &eth::Block, events: &mut contract::Events) {
-    events.poolregistry_pool_addeds.append(&mut blk
-        .receipts()
-        .flat_map(|view| {
-            view.receipt.logs.iter()
-                .filter(|log| log.address == POOLREGISTRY_TRACKED_CONTRACT)
-                .filter_map(|log| {
-                    if let Some(event) = abi::poolregistry_contract::events::PoolAdded::match_and_decode(log) {
-                        return Some(contract::PoolregistryPoolAdded {
-                            evt_tx_hash: Hex(&view.transaction.hash).to_string(),
-                            evt_index: log.block_index,
-                            evt_block_time: Some(blk.timestamp().to_owned()),
-                            evt_block_number: blk.number,
-                            pool: event.pool,
-                            rate_method_id: event.rate_method_id,
-                        });
-                    }
+const POOLREGISTRY_CONTRACT: [u8; 20] = hex!("90e00ace148ca3b23ac1bc8c240c2a7dd9c2d7f5");
```

Comment:
It might be worth looking into parameterization for these so the substream can be ran on other networks all from the cli.

Reply:
I've raised a task for this on the asana project board. For now I've just hardcoded the contract addresses but will look into parameterisation for multi-network support.

Reply:
Besides this, is it good to merge. Have some new changes in a WIP PR that I will finalise Sunday or Monday!